### ISSUE TITLE
imperfect precision but perfect mAPs

### ISSUE THREAD

**Issue by sammilei** [ultralytics/yolov5#1493]
## ❔Question
Hi, I am fine-tuning a single class model with yolov5l.pt and confused by the high mAP at all the testing sets. The model did not perform perfectly on its validation set but it shows perfect scores as 1 or .0999 at mAP.5 and mAP.5:0.95 at test sets even if precision is not 1. Also, the label was almost perfect too. A lot of the time, they perfectly overlay with my labels.

## Additional context
one test result:
!python test.py --weights "/content/yolov5/runs/train/exp/weights/best.pt" --data /content/ms.yaml --img 416 --task test --save-txt --verbose --conf-thres 0.5 --save-conf
```
Namespace(augment=False, batch_size=32, conf_thres=0.5, data='/content/ms.yaml', device='', exist_ok=False, img_size=416, iou_thres=0.6, name='exp', project='runs/test', save_conf=True, save_json=False, save_txt=True, single_cls=False, task='test', verbose=True, weights=['/content/yolov5/runs/train/2_backpack_br_dk/weights/best.pt'])
Using torch 1.7.0+cu101 CUDA:0 (Tesla V100-SXM2-16GB, 16130MB)

Fusing layers... 
Model Summary: 400 layers, 47790077 parameters, 0 gradients
Scanning labels /content/tunnel_backpack_s120_s20000_img/labels/val.cache (590 found, 0 missing, 0 empty, 0 duplicate, for 590 images): 590it [00:00, 17918.67it/s]
               Class      Images     Targets           P           R      mAP@.5  mAP@.5:.95: 100% 19/19 [00:06<00:00,  3.05it/s]
                 all         590         590       0.958           1           1           1
              person         590         590       0.958           1           1           1
Speed: 1.4/2.1/3.4 ms inference/NMS/total per 416x416 image at batch-size 32
Results saved to runs/test/exp14
590 labels saved to runs/test/exp14/labels
```
training plot:
![results](https://user-images.githubusercontent.com/16831346/100065185-5532fe80-2de8-11eb-8166-431dc1fb33cd.png)

Does anyone know what happened? Anything I did wrong?

**Comment by glenn-jocher** [comment]
@sammilei P and R are computed at conf 0.1:
https://github.com/ultralytics/yolov5/blob/7aeef2dca518916a13b7e516505a87c2e77155fd/utils/metrics.py#L41

mAP is computed at conf 0.001:
https://github.com/ultralytics/yolov5/blob/7aeef2dca518916a13b7e516505a87c2e77155fd/test.py#L287

**Comment by sammilei** [comment]
@glenn-jocher Thanks for the quick response. I already added conf-thres using flag --conf-thres 0.5. 
And the system returns:
```
Namespace(augment=False, batch_size=32, conf_thres=0.5, data='/content/ms.yaml', device='', exist_ok=False, img_size=416, iou_thres=0.5, name='exp', project='runs/test', save_conf=True, save_json=False, save_txt=True, single_cls=False, task='test', verbose=True, weights=['/content/train_80_20_large/weights/best.pt'])
Using torch 1.7.0+cu101 CUDA:0 (Tesla V100-SXM2-16GB, 16130MB)
```
Is the flag good enough? I can also try to change the code.

**Comment by sammilei** [comment]
I see what happened. My bad. Thanks, @glenn-jocher for your kind reminder!

**Comment by bonorico** [comment]
I think this might be also related to this issue https://github.com/ultralytics/yolov5/issues/1563, since I see u flagged save-txt in zour command ....
